core/regression_test/regression_test.jl

```diff
@@ -0,0 +1,252 @@
+    config = Ribasim.Config(toml_path)
+
+    flow_bytes_bench =
+        read(normpath(@__DIR__, "../../benchmark/subnetworks_with_sources/flow.arrow"))
+    basin_bytes_bench =
+        read(normpath(@__DIR__, "../../benchmark/subnetworks_with_sources/basin.arrow"))
+    flow_bench = Arrow.Table(flow_bytes_bench)
+    basin_bench = Arrow.Table(basin_bytes_bench)
+
+    solver_list = ["QNDF"]
+    # false sparse or autodiff can cause large differences in result, thus removed
+    sparse_on = [true]
+    autodiff_on = [true]
+
+    @testset "$solver" for solver in solver_list
+        @testset "sparse density is $sparse_on_off" for sparse_on_off in sparse_on
+            @testset "auto differentiation is $autodiff_on_off" for autodiff_on_off in
+                                                                    autodiff_on
+                config = Ribasim.Config(
+                    toml_path;
+                    solver_algorithm = solver,
+                    solver_sparse = sparse_on_off,
+                    solver_autodiff = autodiff_on_off,
+                )
+                model = Ribasim.run(config)
+                @test model isa Ribasim.Model
+                @test successful_retcode(model)
+                (; p) = model.integrator
+
+                # read all results as bytes first to avoid memory mapping
+                # which can have cleanup issues due to file locking
+                flow_bytes = read(normpath(dirname(toml_path), "results/flow.arrow"))
+                basin_bytes = read(normpath(dirname(toml_path), "results/basin.arrow"))
+
+                flow = Arrow.Table(flow_bytes)
+                basin = Arrow.Table(basin_bytes)
+
+                # Testbench for flow.arrow
+                @test flow.time == flow_bench.time
+                @test flow.edge_id == flow_bench.edge_id
+                @test flow.from_node_type == flow_bench.from_node_type
+                @test flow.from_node_id == flow_bench.from_node_id
+                @test flow.to_node_type == flow_bench.to_node_type
+                @test flow.to_node_id == flow_bench.to_node_id
+                @test all(q -> abs(q) < 0.1, flow.flow_rate - flow_bench.flow_rate)
+
+                # Testbench for basin.arrow
+                @test basin.time == basin_bench.time
+                @test basin.node_id == basin_bench.node_id
+                # TODO @test all(q -> abs(q) < 12.0, basin.storage - basin_bench.storage)
+                # TODO @test all(
+                #     q -> abs(q) < 100.0,
+                #     basin.storage[(end - 16):end] - basin_bench.storage[(end - 16):end],
+                # )
+                # TODO @test all(
+                #     q -> abs(q) < 2.0,
+                #     basin.level[(end - 16):end] - basin_bench.level[(end - 16):end],
+                # )
+                # TODO @test all(q -> abs(q) < 1e-1, basin.balance_error)
+                # TODO @test all(q -> abs(q) < 30.0, basin.relative_error)
```

Comment:
Don't forget to update.

Reply:
Allocation would return different stretegy if running the same model twice. This is because some optimization problem has unlimit solutions and the solver returns a random solution. Therefore the output of allocation model is always different. This cannot be updated until allocation can return same strategy in different run. I will deleted them for now, but this should be investigated.